src/hotspot/cpu/x86/sharedRuntime_x86_64.cpp

```diff
@@ -234,11 +234,12 @@ OopMap* RegisterSaver::save_live_registers(MacroAssembler* masm, int additional_
     }
   } else {
     if (VM_Version::supports_evex()) {
-      // Save upper bank of ZMM registers(16..31) for double/float usage
+      // Save upper bank of XMM registers(16..31) for scalar or 16-byte vector usage
```

Comment:
I have a question:
This is on the else path where save_wide_vectors is false. Why do we need to save the full vector registers here? 
Could this be a problem at the call site of this method (save_live_registers) from where save_wide_vectors is not being sent as true?

Reply:
C2 uses SharedRuntime::is_wide_vector(C->max_vector_size()) to determine the value of save_wide_vectors, and is_wide_vector() does
`return size > 16;`
so we need to save vector bytes 0 - 15 even if save_wide_vectors is false.